### PR TITLE
[DEV-6449] API Integration for 1st Tab on Overview Page

### PR DESCRIPTION
**Pull request by MaxwellKendall**
Contains #2295 

**High level description:**
Integrates with the API & Includes a few fixes surfaced by it.

**Technical details:**
8621101 - API Integration + Fix related to Pagination Total Items
8b1cec3 - Styling fix for loading

**JIRA Ticket:**
[DEV-6449](https://federal-spending-transparency.atlassian.net/browse/DEV-6449)

The following are ALL required for the PR to be merged:

Author:
- [x] Linked to this PR in JIRA ticket
`N/A` Scheduled demo including Design/Testing/Front-end OR Provided instructions for testing in JIRA and PR `if applicable`
`N/A` Verified cross-browser compatibility: Chrome, Safari, Firefox, Internet Explorer, Edge
`N/A` Verified mobile/tablet/desktop/monitor responsiveness
`N/A` Verified that this PR does not create any *new* accessibility issues (via Axe Chrome extension)
`N/A` Added Unit Tests for helper functions, reducers, models and Container/Component Interactivity Expectations `if applicable` [React Testing Library](react-testing-library.md)
`N/A` [API contract](https://github.com/fedspendingtransparency/usaspending-api/tree/dev/usaspending_api/api_contracts) updated `if applicable`
`N/A` [Component Library Integration Status Report](https://github.com/fedspendingtransparency/data-act-documentation/blob/data-transparency-ui/frontend_apps/component-library-integration-status.md) updated `if applicable`

Reviewer(s):
`N/A` Design review complete `if applicable`
`N/A` [API #1234](https://github.com/fedspendingtransparency/usaspending-api/pull/1234) merged concurrently `if applicable`
- [x] Code review complete
- [x] #2295 reviewed

### CODE REVIEW

File: src/js/helpers/aboutTheDataHelper.js [hunk 1]
```diff
@@ -63,12 +63,13 @@ export const getLastPeriodWithinQuarterByPeriod = (periodId) => (
 
 const defaultState = {
     page: 1,
-    limit: 10
+    limit: 10,
+    totalItems: 1
```
Comment (jonhill13) [inline src/js/helpers/aboutTheDataHelper.js:67]: Why default `totalItems` to 1?

File: src/js/models/v2/aboutTheData/ReportingOverviewRow.js [hunk 2]
```diff
@@ -19,6 +19,8 @@ const BaseAgencyRow = {
         this.certified = data.recent_publication_date_certified || false;
         this.tasTotals = data.tas_account_discrepancies_totals;
         this._federalTotal = data.federalTotal;
+        this.unlinkedContractAwards = data.unlinked_contract_award_count;
```
Comment (jonhill13) [inline src/js/models/v2/aboutTheData/ReportingOverviewRow.js:22]: are we handling no data for these two? I guess the api could be responding with 0 if there is no data

**Review by jonhill13 — APPROVED** [review]

**Comment by jonhill13** [comment]
@MaxwellKendall these two routes seem to be the same route?
Screen Shot 2021-01-07 at 12 20 49 PM

File: src/js/helpers/aboutTheDataHelper.js [hunk 3]
```diff
@@ -94,37 +95,16 @@ export const getTotalBudgetaryResources = (fy, period) => {
...
-                });
-            }, 500);
-        }),
-        cancel: () => {
-            console.log('cancel executed!');
-        }
-    };
-};
+export const getAgenciesReportingData = (fy, period, sort, order, page, limit) => apiRequest({
+    url: `v2/reporting/agencies/overview?${stringify({
```
Comment (MaxwellKendall) [inline src/js/helpers/aboutTheDataHelper.js:99]:
```suggestion
    url: `v2/reporting/agencies/overview/?${stringify({
```

File: src/js/helpers/aboutTheDataHelper.js (hunk 1 above)
Comment (MaxwellKendall) [reply src/js/helpers/aboutTheDataHelper.js:67]: good question

**Pull request merged**